### DISCUSSION TITLE
MagickResizeImage results in a strange image

### DISCUSSION THREAD

**Discussion by preoteasa** [ImageMagick/ImageMagick#2140] [Help]
I have recompiled ImageMagick recently on Windows using static libraries. When I use MagickResizeImage it results in the attached image. 

![error](https://user-images.githubusercontent.com/6583978/84577096-078a8280-adc2-11ea-9448-32a951e186db.jpg)
The original image is a png with transparent background showing a mug. Just replacing the transparency with white color and saving as jpg works OK. Any idea of what may be wrong? To resize the image I used:

MagickResizeImage(m_wand, width, height, LanczosFilter, 1)

Before recompiling the new sources I have used the same code without problems for many years.

**Comment by fmw42** [comment]
I do not understand the issue.  What is wrong? Can you provide the input image. Also please identify your exact IM version.  What should it look like?

**Reply by preoteasa** [reply]
Here is the original image. Where do I find the version in the sources?
![Muurla Moomin In the garden, Girls enamel mug 2,5dl 1711-025-03 6416114964727 1](https://user-images.githubusercontent.com/6583978/84577304-d8751080-adc3-11ea-84d4-b21fd47b8be6.png)

**Reply by preoteasa** [reply]
It is version: ImageMagick-7.0.10-10

**Reply by fmw42** [reply]
Thanks. But I do not understand. You got that green image from the pink mug? What was your exact command with arguments? I am not on Windows, but will try to reproduce in command line. Have you tried to upgrade? IM 7 is now at 7.0.10.19

**Reply by preoteasa** [reply]
Yes, that is my problem. For some images you can recognize the object. But not only that I get that strange image, it is also duplicated. The ratio width/height is twice as big in the resized picture, although the ratio should be the same.

**Reply by preoteasa** [reply]
![a](https://user-images.githubusercontent.com/6583978/84577509-86cd8580-adc5-11ea-91a8-2939e26bc7d7.jpg)
![localhost](https://user-images.githubusercontent.com/6583978/84577510-892fdf80-adc5-11ea-9138-a88b5f563019.jpg)


Here you can see better the original and the resized image. In the resized image you can recognize the original twice. The original is jpg in this case.

**Reply by fmw42** [reply]
I cannot reproduce that in command line?  Does it happen if you use -resize in command line mode?

**Reply by fmw42** [reply]
Which image is the original?

**Reply by preoteasa** [reply]
I tested now the command line compiled by myself, and it resulted in the same problem:

convert logo: image.jpg
convert -resize 100 image.jpg image-small.jpg

![image](https://user-images.githubusercontent.com/6583978/84577624-a87b3c80-adc6-11ea-98bb-88eeb18b1be0.jpg)
![image-small](https://user-images.githubusercontent.com/6583978/84577626-ab762d00-adc6-11ea-9522-033f372f9d72.jpg)

**Reply by preoteasa** [reply]
On the ImageMagick site there seem to be a newer version 7.0.10-19. So maybe that does not have this problem. I will try that as well.

**Reply by fmw42** [reply]
Try reading the input before the -resize.

```
convert logo: image.jpg
convert image.jpg -resize 100 image-small.jpg

```

If you are using IM 7, then replace convert with magick

**Reply by fmw42** [reply]
What do you get from either:

convert -version

or 

magick -version

**Reply by preoteasa** [reply]
It is the same result. The C API function gives the same kind of result, so it must be something else.

**Reply by preoteasa** [reply]
Version:

Version: ImageMagick 7.0.10-10 Q32 x64 2020-05-03 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 191627039
Features: Cipher DPC HDRI OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib

**Comment by preoteasa** [comment]
I just compiled ImageMagick-7.0.10-19. It seems that it works. So problem solved.

**Reply by preoteasa** [reply]
I notice now that I have Q32 in 7.0.10-10. Now I left default Q16.

**Comment by fmw42** [comment]
Glad to hear that.

**Reply by preoteasa** [reply]
Could it be that Q32 is causing the problem?

**Reply by fmw42** [reply]
Not likely, but a possibility. But you are now on Q32 and it works. I doubt that the problem would be on Q16 as I tested that and it works on my platform.  Did you install the proper Windows version on your computer?  Be sure you install a 64-bit version if your computer is 64-bit Windows and 32-bit version if you are on an older computer that on has 32-bit Windows.

**Reply by preoteasa** [reply]
I have chosen Q16 for the new compilation. I compiled ImageMagick from sources as I need the static libraries. And I compiled 64-bit version because my application is 64 bit. So all is good. There is something wrong in ImageMagick-7.0.10-10 (Q32). I am using now ImageMagick-7.0.10-19 (Q16).

**Reply by fmw42** [reply]
OK. I will have to leave that now for the IM developers to explore further.
